src/rgw/rgw_data_sync.cc

```diff
@@ -1337,21 +1342,26 @@ class RGWDataSyncShardCR : public RGWCoroutine {
           } else {
             // fetch remote and write locally
             yield spawn(new RGWDataSyncSingleEntryCR(sync_env, *iter, *iter, marker_tracker, error_repo, false, tn), false);
```

Comment:
any reason why not looking at retcode here? what if spawn fails? (not sure it can actually fail, but shouldn't we check it anyway?)

Reply:
if the spawned coroutine fails, we have to collect() it to inspect the error code. my understanding has been that call() will set the retcode and needs to be checked, but spawn() won't - and most of the other code in rgw_sync/data_sync.cc seem to be consistent in only checking on call()

Reply:
@cbodley I'm worried about a coroutine that fails to spawn, so collect doesn't have it. But since it's just a theoretical problem and other callers don't do it then it's ok for now. Will need to revise if `spawn()` actually does end up setting retcode.